feature_engine/datetime/datetime.py

```diff
@@ -250,13 +271,32 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         # Check if input data contains same number of columns as dataframe used to fit.
         _check_input_matches_training_df(X, self.n_features_in_)
 
+        # special case index
+        if self.variables_ is None:
+            if self.missing_values == "raise":
+                self._check_index_contains_na(X.index)
+
+            for feat in self.features_to_extract_:
+                X[FEATURES_SUFFIXES[feat][1:]] = FEATURES_FUNCTIONS[feat](
+                    pd.Series(
+                        pd.to_datetime(
+                            X.index,
+                            dayfirst=self.dayfirst,
+                            yearfirst=self.yearfirst,
+                            utc=self.utc,
+                        ),
+                        index=X.index,
+                    )
+                )
+
+            return X
```

Comment:
could we have just 1 return at the end of the method and not break the logic in 2?

Reply:
If you'd rather have what follows under an else statement yea we can do that